x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -97,9 +98,27 @@ export function App({
         toDate: currentRange.to,
       },
       filters: [],
+      indicateNoData: false,
     };
   });
 
+  const showNoDataPopover = useCallback(() => {
+    setState((prevState) => ({ ...prevState, indicateNoData: true }));
+  }, [setState]);
+
+  useEffect(() => {
+    if (state.indicateNoData) {
+      setState((prevState) => ({ ...prevState, indicateNoData: false }));
+    }
+  }, [
+    setState,
+    state.indicateNoData,
+    state.query,
```

Comment:
Why did you have to list attributes that are not used explicitly in the effect? Is it because of prevState? :thinking:

Reply:
We want to show this popover only on first load - if the user starts to change the configuration, they most likely now how to get data (e.g. by changing the index pattern because the wrong one was pre-selected). Then we don't need the popover anymore.

This is just to be extra save (as the popover is also hidden when the user interacts with the UI outside of it)

Reply:
Might be nice adding a comment on this as this is not trivial behavior